### PR TITLE
moving out to kts scripts

### PR DESCRIPTION
**Pull request by operasfantom**

### CODE REVIEW

File: buildSrc/src/main/kotlin/com/jetbrains/rd/gradle/tasks/GenerateTask.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        main = "com.jetbrains.rd.generator.nova.MainKt"
+        outputs.upToDateWhen { true }
+    }
+
+    override fun exec() {
+        super.exec()
+
+        (systemProperties["model.out.src.cpp.dir"] as? String)?.let { output ->
+/*
+            project.copy {
```
Comment (korifey) [inline buildSrc/src/main/kotlin/com/jetbrains/rd/gradle/tasks/GenerateTask.kt]: maybe remove commented?

File: buildSrc/src/main/kotlin/com/jetbrains/rd/gradle/tasks/GenerateTask.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    init {
+        group = "generate"
+        main = "com.jetbrains.rd.generator.nova.MainKt"
+        outputs.upToDateWhen { true }
+    }
+
+    override fun exec() {
+        super.exec()
+
+        (systemProperties["model.out.src.cpp.dir"] as? String)?.let { output ->
```
Comment (korifey) [inline buildSrc/src/main/kotlin/com/jetbrains/rd/gradle/tasks/GenerateTask.kt]: is this task solely about cpp?

File: rd-kt/rd-cross/cross.gradle.kts [hunk 3]
```diff
@@ -1,94 +1,94 @@
...
 def getCppTaskByName(String name) {
-    return project(':rd-cpp').getTasksByName(name, true).iterator().next()
+    return project(":rd-cpp").getTasksByName(name, true).iterator().next()
 }
 */
 
-def getCsTaskByName(String name) {
-    return project(':rd-net').getTasksByName(name, true).iterator().next()
-}
+/*fun getCsTaskByName(name : String): Task {
```
Comment (korifey) [inline rd-kt/rd-cross/cross.gradle.kts]: why such a lot of commented strings?

File: rd-kt/rd-cross/cross.gradle.kts [hunk 4]
```diff
@@ -1,94 +1,94 @@
...
+    taskClient = getCsTaskByName("CrossTestCsClientRdCall")
 
     addDependencies()
-}
+}*/
 //endregion
 
-def interopTasks = [/*CrossTestKtCppAllEntities, CrossTestKtCppBigBuffer, CrossTestKtCppRdCall,*/
-                    CrossTestKtCsAllEntities, CrossTestKtCsBigBuffer, CrossTestKtCsRdCall]
+//val interopTasks = listOf(/*CrossTestKtCppAllEntities, CrossTestKtCppBigBuffer, CrossTestKtCppRdCall,*/
```
Comment (korifey) [inline rd-kt/rd-cross/cross.gradle.kts]: please remove everything commented

File: rd-kt/rd-framework/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        commonTest {
+
+        }
+
+    }
+}
+
+sourceSets {
+    crossTest = create("crossTest") {
+//        compileClasspath += sourceSets.main.output + configurations.testRuntimeClasspath
```
Comment (korifey) [inline rd-kt/rd-framework/build.gradle.kts]: commented

File: rd-kt/rd-framework/src/crossTest/kotlin/com/jetbrains/rd/cross/CrossTestKtServerAllEntities.kt [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+            printer.printIfRemoteChange(model.set, "set", e, x)
+        }
+
+        model.mapLongToString.advise(lifetime) {
+            printer.printIfRemoteChange(model.mapLongToString, "mapLongToString", it)
+        }
+
+/*
+    model.call.set { c ->
+        printer.print("RdTask:")
```
Comment (korifey) [inline rd-kt/rd-framework/src/crossTest/kotlin/com/jetbrains/rd/cross/CrossTestKtServerAllEntities.kt]: comment

File: rd-kt/rd-gen/models.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    val generateInterningTestModel by creating(GenerateTask::class) {
+        initializeClasspath()
+
+        sourcesRoot = ktRoot.resolve(modelsRelativePath)
+        sourcesFolder = "interning"
+
+        systemProperties = mapOf(
+                "model.out.src.cpp.dir" to "$cppRoot/src/rd_framework_cpp/src/test/util/interning",
+                "model.out.src.kt.dir" to "$BUILD_DIR/models/interning"
+//            "model.out.src.cs.dir" : "$csRoot/"
```
Comment (korifey) [inline rd-kt/rd-gen/models.gradle.kts]: comment

File: rd-net/CrossTest/TestRunner.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Test.RdCross
+{
+    public class TestRunner
+    {
+        public static void Main(string[] args)
+        {
+            if (args.Length < 1)
+            {
+                throw new ArgumentException($"Wrong number of arguments:${args.Length}");
```
Comment (korifey) [inline rd-net/CrossTest/TestRunner.cs]: Throw better diagnostics. What is expected number of arguments and what is the argument. Example would be perfect.

File: rd-net/CrossTest/TestRunner.cs [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+            }
+            var type = Type.GetType($"Test.RdCross.{args[0]}") ??
+                       throw new ArgumentException($"Wrong class name={args[0]})");
+            if (Activator.CreateInstance(type) is CrossTestCsBase testCsBase)
+            {
+                testCsBase.Run(args.Skip(1).ToArray());
+            }
+            else
+            {
+                throw new ArgumentException(" ");
```
Comment (korifey) [inline rd-net/CrossTest/TestRunner.cs]: Exception with empty message?!

File: buildSrc/src/main/kotlin/com/jetbrains/rd/gradle/tasks/GenerateTask.kt (hunk 1 above)
Comment (operasfantom) [reply buildSrc/src/main/kotlin/com/jetbrains/rd/gradle/tasks/GenerateTask.kt]: Will fix it

File: buildSrc/src/main/kotlin/com/jetbrains/rd/gradle/tasks/GenerateTask.kt (hunk 2 above)
Comment (operasfantom) [reply buildSrc/src/main/kotlin/com/jetbrains/rd/gradle/tasks/GenerateTask.kt]: No, it's the same as RdgenTask

**Comment by operasfantom** [comment]
Is everything okay now?

File: rd-net/CrossTest/CrossTestCsClientAllEntities.cs [hunk 10]
```diff
@@ -42,7 +38,7 @@ private void AdviseAll(Lifetime lifetime, DemoModel model, ExtModel extModel)
                 Printer.Print(s);
 
                 return 50;
-            });*/
+            });#1#
```
Comment (korifey) [inline rd-net/CrossTest/CrossTestCsClientAllEntities.cs]: what is it?

Comment (operasfantom) [reply rd-net/CrossTest/CrossTestCsClientAllEntities.cs]: Nested comments in C#

**Pull request merged**